include/boost/thread/pthread/mutex.hpp

```diff
@@ -34,6 +36,10 @@
 #endif
 #endif
 
+// CLOCK_MONOTONIC only works with pthread_cond_timedwait(), not with pthread_mutex_timedlock()
+#ifdef BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC
+#undef BOOST_PTHREAD_HAS_TIMEDLOCK
```

Comment:
We can not undef this here. This is a global define. If Boost.Thread can not use pthread_mutex_timedlock() when BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC is defined, this should stay internal to Boost.Thread

Reply:
Ah, good to know. I'll have to fix that.

Reply:
I believe it is enough to check in addition  is defined or not BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC or define a new specific BOOST_THREAD_MACRO.